### PR TITLE
Android test variant support

### PR DESCRIPTION
**Pull request by Tanish-Ranjan**
Addressing https://github.com/microsoft/build-server-for-gradle/issues/183, this PR discovers all the test variants and adds them to the source sets.

### CODE REVIEW

**Comment by Tanish-Ranjan** [comment]
This has been branched from android-support branch. I will rebase once https://github.com/microsoft/build-server-for-gradle/pull/173 is merged.

**Comment by Tanish-Ranjan** [comment]
Tested the changes with default test project by integrating with [vscode-gradle](https://github.com/microsoft/vscode-gradle) extension. The test variants were retrieved successfully.

File: plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/utils/AndroidUtils.java [hunk 1]
```diff
@@ -156,8 +163,57 @@ private static GradleSourceSet convertVariantToGradleSourceSet(Project project,
...
+      addGeneratedSourceAndSourceOutputs(gradleSourceSet, variant, compilerArgs);
+
+      // classpath
+      addClasspath(gradleSourceSet, variant);
+
+      // Archive output dirs (not relevant in case of android build variants)
+      gradleSourceSet.setArchiveOutputFiles(new HashMap<>());
+
+      // has tests
+      addTests(gradleSourceSet, project, variant, sourceSets);
```
Comment (jdneo) [inline plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/utils/AndroidUtils.java:186]: Why `addTests()` put inside `convertVariantToGradleSourceSet()`? Can we handle them separately?

File: plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/utils/AndroidUtils.java [hunk 2]
```diff
@@ -116,9 +117,15 @@ public static List<GradleSourceSet> getBuildVariantsAsGradleSourceSets(Project p
    *
    * @param project Gradle project to populate GradleSourceSet properties
    * @param variant Android Build Variant object to populate GradleSourceSet properties
+   * @param sourceSets List of source sets to which test variants will be added
+   * @param isUnitTest Indicates if the given variant is a unit test variant
    */
-  @SuppressWarnings("unchecked")
-  private static GradleSourceSet convertVariantToGradleSourceSet(Project project, Object variant) {
+  private static GradleSourceSet convertVariantToGradleSourceSet(
```
Comment (jdneo) [inline plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/utils/AndroidUtils.java:143]: We can change the function to return a collection of source set, and add them out of the function

Comment (Tanish-Ranjan) [reply plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/utils/AndroidUtils.java:143]: Resolved in https://github.com/microsoft/build-server-for-gradle/pull/194/commits/736a14ca181bcf75ac0c11fa65d4b691192cf3c8

**Review by jdneo — APPROVED** [review]

**Comment by jdneo** [comment]
Thank you for your contribution! @Tanish-Ranjan

**Pull request merged**